secrets/localsecrets/localsecrets.go

```diff
@@ -106,16 +110,39 @@ func Base64Key(base64str string) ([32]byte, error) {
 	if err != nil {
 		return sk32, err
 	}
+	keySize := len([]byte(key))
+	if keySize != 32 {
+		return sk32, fmt.Errorf("Base64Key: secret material is %v bytes, want 32 bytes", keySize)
+	}
 	copy(sk32[:], key)
 	return sk32, nil
 }
 
 // ByteKey takes a secret key as a string and converts it
-// to a [32]byte, cropping it if necessary.
-func ByteKey(sk string) [32]byte {
+// to a [32]byte, erroring if the string is not 32 bytes of data.
+func ByteKey(sk string) ([32]byte, error) {
+	var sk32 [32]byte
+	skb := []byte(sk)
+	keySize := len(skb)
+	if keySize != 32 {
+		return sk32, fmt.Errorf("ByteKey: secret material is %v bytes, want 32 bytes", keySize)
+	}
+	copy(sk32[:], skb)
+	return sk32, nil
+}
+
+// ThirtyTwoByteSecret will generate random secret material suitable to be
+// used as the secret key argument to NewKeeper.
+func ThirtyTwoByteSecret() ([32]byte, error) {
 	var sk32 [32]byte
-	copy(sk32[:], []byte(sk))
-	return sk32
+	randomBytes := make([]byte, 32)
+	// Read random numbers into the passed slice until it's full
+	_, err := rand.Read(randomBytes)
+	if err != nil {
+		return sk32, err
```

Comment:
https://golang.org/pkg/math/rand/#Rand.Read says it always returns a `nil` error, so I think you can ignore the error here (or panic), and drop the `error` return value from this function.

Reply:
ah, this is from [crypto/rand](https://godoc.org/crypto/rand#Read), rather than `math/rand`, which seems to indicate it sometimes returns a real error.

Reply:
Yup, you're right, never mind.